src/main/webapp/js/maps-page-loader.js

```diff
@@ -39,5 +44,85 @@ function initConfigMap() {
     // Associate the styled map with the MapTypeId and set it to display.
     map.mapTypes.set(MAP_NIGHT_TYPE_ID, styledMapType);
     map.setMapTypeId(MAP_NIGHT_TYPE_ID);
+
+    // When the user clicks in the map, show a marker with a text box the user can edit.
+    map.addListener('click', (event) => {
+      createMarkerForEdit(event.latLng.lat(), event.latLng.lng());
+    });
+    fetchMarkers();
+  });
+}
+
+/** Fetches markers from the backend and adds them to the map. */
+function fetchMarkers(){
+  fetch('/markers').then((response) => {
+    return response.json();
+  }).then((markers) => {
+    markers.forEach((marker) => {
+     createMarkerForDisplay(marker.lat, marker.lng, marker.content)
+    });
+  });
+}
+
+/** Creates a marker that shows a read-only info window when clicked. */
+function createMarkerForDisplay(lat, lng, content){
+  const marker = new google.maps.Marker({
+    position: {lat: lat, lng: lng},
+    map: map
   });
+  var infoWindow = new google.maps.InfoWindow({
+    content: content
+  });
+  marker.addListener('click', () => {
+    infoWindow.open(map, marker);
+  });
+}
+
+/** Sends a marker to the backend for saving. */
+function postMarker(lat, lng, content){
+  const params = new URLSearchParams();
+  params.append('lat', lat);
+  params.append('lng', lng);
+  params.append('content', content);
+  fetch('/markers', {
+    method: 'POST',
+    body: params
+  });
+}
+
+/** Creates a marker that shows a textbox the user can edit. */
+function createMarkerForEdit(lat, lng){
+  // If an editable marker is already shown, remove it.
+  if (editMarker) {
+   editMarker.setMap(null);
+  }
+  editMarker = new google.maps.Marker({
+    position: {lat: lat, lng: lng},
```

Comment:
Depending on the javascript version:
```suggestion
    position: {lat, lng},
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#New_notations_in_ECMAScript_2015

Reply:
Is it more desirable to be backward js version compatible?